cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -1290,14 +1514,122 @@ instance Show peerConn => ToObject (DebugPeerSelection Socket.SockAddr peerConn)
+    mkObject [ "kind" .= String "PeerMonitoringError"
+             , "connectionId" .= toObject _verb cs
+             , "error" .= show s
+             ]
+  toObject _verb (PeerMonitoringResult cs wf) =
+    mkObject [ "kind" .= String "PeerMonitoringResult"
+             , "connectionId" .= toObject _verb cs
+             , "withProtocolTemp" .= show wf
+             ]
 
 instance ToObject PeerSelectionCounters where
   toObject _verb ev =
     mkObject [ "kind" .= String "PeerSelectionCounters"
-             , "event" .= show ev ]
+             , "coldPeers" .= coldPeers ev
+             , "warmPeers" .= warmPeers ev
+             , "hotPeers" .= hotPeers ev
+             ]
+
+instance (Show (ClientHasAgency st), Show (ServerHasAgency st))
+  => ToJSON (PeerHasAgency pr st) where
+  toJSON (ClientAgency cha) =
+    Aeson.object [ "kind" .= String "ClientAgency"
+                 , "agency" .= show cha
+                 ]
+  toJSON (ServerAgency sha) =
+    Aeson.object [ "kind" .= String "ServerAgency"
+                 , "agency" .= show sha
+                 ]
+
+instance ToJSON ProtocolLimitFailure where
+  toJSON (ExceededSizeLimit tok) =
+    Aeson.object [ "kind" .= String "ProtocolLimitFailure"
+                 , "agency" .= toJSON tok
+                 ]
+  toJSON (ExceededTimeLimit tok) =
+    Aeson.object [ "kind" .= String "ProtocolLimitFailure"
+                 , "agency" .= toJSON tok
+                 ]
+
+instance Show vNumber => ToJSON (RefuseReason vNumber) where
+  toJSON (VersionMismatch vNumber tags) =
+    Aeson.object [ "kind" .= String "VersionMismatch"
+                 , "versionNumber" .= show vNumber
+                 , "tags" .= show tags
+                 ]
+  toJSON (HandshakeDecodeError vNumber t) =
+    Aeson.object [ "kind" .= String "HandshakeDecodeError"
+                 , "versionNumber" .= show vNumber
+                 , "text" .= show t
+                 ]
+  toJSON (Refused vNumber t) =
+    Aeson.object [ "kind" .= String "Refused"
+                 , "versionNumber" .= show vNumber
+                 , "text" .= show t
+                 ]
+
+instance Show vNumber => ToJSON (HandshakeProtocolError vNumber) where
+  toJSON (HandshakeError rvNumber) =
+    Aeson.object [ "kind" .= String "HandshakeError"
+                 , "versionNumber" .= toJSON rvNumber
+                 ]
+  toJSON (NotRecognisedVersion vNumber) =
+    Aeson.object [ "kind" .= String "NotRecognisedVersion"
+                 , "versionNumber" .= show vNumber
+                 ]
+  toJSON (InvalidServerSelection vNumber t) =
+    Aeson.object [ "kind" .= String "InvalidServerSelection"
+                 , "versionNumber" .= show vNumber
+                 , "reason" .= t
+                 ]
+
+instance Show vNumber => ToJSON (HandshakeException vNumber) where
+  toJSON (HandshakeProtocolLimit plf) =
+    Aeson.object [ "kind" .= String "HandshakeProtocolLimit"
+                 , "handshakeProtocolLimit" .= show plf
```

Comment:
we have a `ToJSON ProtocolLimitFailure` instance which we can benefit from here.